client/onboarding/add-business-info-task/index.js

```diff
@@ -0,0 +1,171 @@
+			};
+		}
+	);
+
+	const [ businessCountry, setBusinessCountry ] = useState( country );
+	const [ businessType, setBusinessType ] = useState();
+	const [ businessStructure, setBusinessStructure ] = useState();
+	const [ displayStructures, setDisplayStructures ] = useState( false );
+
+	const getBusinessTypesForCountry = ( countryCode ) => {
+		if ( isLoading || ! businessTypes.hasOwnProperty( 'data' ) ) {
+			return false;
+		}
+
+		return businessTypes.data.hasOwnProperty( countryCode )
+			? businessTypes.data[ countryCode ]
+			: false;
+	};
+
+	const formatBusinessTypes = ( countryCode ) => {
+		const businessTypeData = getBusinessTypesForCountry( countryCode );
+
+		if ( isLoading || ! businessTypeData ) {
+			return [];
+		}
+
+		return Object.keys( businessTypeData ).map( ( key ) => ( {
+			name: strings.businessTypes[ key ],
+			description: 'Test',
+			key: key,
+		} ) );
+	};
+
+	const shouldDisplayStructures = ( countryCode, type ) => {
+		const businessTypeData = getBusinessTypesForCountry( countryCode );
+
+		if ( isLoading || ! businessTypeData ) {
+			return false;
+		}
+
+		return businessTypeData.hasOwnProperty( type )
+			? 0 < businessTypeData[ type ].length
+			: false;
+	};
+
+	const formatBusinessStructures = ( countryCode, type ) => {
+		const businessTypeData = getBusinessTypesForCountry( countryCode );
+
+		if ( isLoading || ! businessTypeData ) {
+			return false;
+		}
+
+		return businessTypeData.hasOwnProperty( type )
+			? businessTypeData[ type ]
+			: [];
+	};
+
+	const handleBusinessTypeUpdate = ( type ) => {
+		setBusinessType( type );
+
+		if ( shouldDisplayStructures( businessCountry, type ) ) {
+			formatBusinessStructures( businessCountry, type );
+			setDisplayStructures( true );
+		} else {
+			setDisplayStructures( false );
+			setCompleted( true );
+		}
+	};
+
+	const handleBusinessStructureUpdate = ( structure ) => {
+		setBusinessStructure( structure );
+		setCompleted( true );
+	};
+
+	const handleBusinessCountryUpdate = ( countryCode ) => {
+		setBusinessCountry( countryCode );
+		formatBusinessTypes( countryCode );
+		setCompleted( false );
+	};
+
+	return (
+		<WizardTaskItem
+			className="complete-business-info-task"
+			index={ 1 }
+			title={ strings.onboarding.heading }
+		>
+			{ isLoading ? <p>Loading...</p> : '' }
+
+			<p className="wcpay-wizard-task__description-element subheading is-muted-color">
+				{ strings.onboarding.description }
+			</p>
+
+			<SelectControl
+				label={ __( 'Country', 'woocommerce-payments' ) }
+				value={ businessCountry }
+				onChange={ ( value ) => handleBusinessCountryUpdate( value ) }
+				options={ countries }
+			/>
+
+			<p className="wcpay-wizard-task__description-element is-muted-color">
+				{ strings.onboarding.countryDescription }
+			</p>
+
+			{ isLoading ? null : (
+				<OnboardingSelectControl
+					className="wcpay-onboarding-select-business-types"
+					label={ __( 'Business type', 'woocommerce-payments' ) }
+					value={ businessType }
```

Comment:
The select doesn't work because we are using the received object as value here. We need to access `selectedItem`. You can for example destructure it with something like:
```jsx
<OnboardingSelectControl onChange={ ( { selectedItem } ) => handleBusinessTypeUpdate( selectedItem ) }>
```

Reply:
Thanks! Resolved in 9bd42017c7ebd3476557100b7ff1175cc5ba2c9b.